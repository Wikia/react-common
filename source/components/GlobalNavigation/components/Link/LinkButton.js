import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import Button from '../../../Button';

const LinkButton = ({ link }) => {
    const [t] = useTranslation();

    return (
        <Button
            className="wds-global-navigation__link-button"
            secondary
            data-tracking-label={link['tracking-label']}
            href={link.href}
        >
            {t(link.title.key)}
        </Button>
    );
};

LinkButton.propTypes = {
    link: PropTypes.shape().isRequired,
};

export default LinkButton;
