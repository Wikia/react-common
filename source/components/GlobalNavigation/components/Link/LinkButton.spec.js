import React from 'react';
import renderer from 'react-test-renderer';
import merge from 'lodash/merge';

import LinkButton from './LinkButton';

const defaultProps = {
    link: {
        type: 'link-button',
        title: {
            type: 'translatable-text',
            key: 'global-navigation-create-wiki-link-start-wikia',
        },
        href: '\/\/www.wikia.com\/Special:CreateNewWiki',
        'tracking-label': 'link.start-a-wiki',
    }
};

function renderComponent(props) {
    const computedProps = merge({}, defaultProps, props);

    return renderer.create(<LinkButton {...computedProps} />);
}

test('LinkButton renders correctly', () => {
    expect(renderComponent().toJSON()).toMatchSnapshot();
});
