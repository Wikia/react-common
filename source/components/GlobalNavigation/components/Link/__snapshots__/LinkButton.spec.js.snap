// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correctly 1`] = `
<Button
  className="wds-global-navigation__link-button"
  data-tracking-label="link.start-a-wiki"
  disabled={false}
  fullwidth={false}
  href="//www.wikia.com/Special:CreateNewWiki"
  onClick={[Function]}
  secondary={true}
  square={false}
  text={false}
>
  global-navigation-create-wiki-link-start-wikia
</Button>
`;
